File: backend/app/Models/Activity.php
<?php

namespace App\Models;

use App\Models\Module;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Activity extends Model
{
    use HasFactory;

    protected $fillable = [
        'module_id',
        'title',
        'content',
        'type',
        'correct_answer',
        'points',
    ];

    public function module()
    {
        return $this->belongsTo(Module::class);
    }

    public function users()
    {
        return $this->belongsToMany(User::class, 'user_activities')
            ->withPivot('answer', 'is_correct', 'is_completed')
            ->withTimestamps();
    }
}

File: app/app/src/main/java/com/example/alfabetize/models/Activity.kt
package com.example.alfabetize.models

data class Activity(
    val id: Int,
    val module_id: Int,
    val title: String,
    val content: String,
    val type: String,
    val correct_answer: String,
    val createdAt: String,
    val updatedAt: String,
    val points: Int,
)